WordPress/Classes/ViewRelated/Me/Domain Details/DomainDetailsWebViewController.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+
+final class DomainDetailsWebViewController: WebKitViewController {
+
+    // MARK: - Properties
+
+    private var observation: NSKeyValueObservation?
+
+    // MARK: - Init
+
+    init(url: URL, analyticsSource: String? = nil) {
+        let configuration = WebViewControllerConfiguration(url: url)
+        configuration.analyticsSource = analyticsSource
+        configuration.secureInteraction = true
+        configuration.authenticateWithDefaultAccount()
+        super.init(configuration: configuration)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.observeURL()
+    }
+
+    // MARK: - Handling URL Changes
+
+    private func observeURL() {
```

Comment:
[Suggestion] Instead of observing the url and navigating back, how about using `WKWebView.navigationDelegate`?
See [BlazeWebViewController.swift](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Classes/ViewRelated/Blaze/Webview/BlazeWebViewController.swift) for reference

Reply:
@hassaanelgarem That was the first thing I've tried, but it didn't catch all URL changes. For example, that method wasn't called when I tapped `Payment Details` button.

I also tried overriding this `WKNavigationDelegate` method:

```swift
func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void)
```

Navigation info can be in these properties:
- `navigationAction.request.url `
- `navigationAction.sourceFrame.url`
- `navigationAction.targetFrame.url`

I tried all those properties but there was still cases where this delegate method doesn't catch all URL changes.